### PR TITLE
8283507: Create a regression test for RFE 4287690

### PR DESCRIPTION
**Pull request by manukumarvs**
Create a regression test for [JDK-4287690](https://bugs.openjdk.java.net/browse/JDK-4287690)
Issue description:
I want JComboBox to send an event just before the popup (drop down) list is displayed.

Fix:
This requires some additional API targeted for major release of the JDK (1.4).The PopupMenuEvent will from the JPopupMenu will be forwarded to the JComboBox. A client of the JComboBox can register itself as a PopupMenuListener to receive the notification.
So JComboBox should send drop down events when the drop down popup is visible as well as invisible. JComboBox.addPopupMenuListener(PopupMenuListener) can be used to register for pop up events.

Testing:
Tested using Mach5 in all platforms available(macos,linux and windows) and got all pass.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8283507](https://bugs.openjdk.java.net/browse/JDK-8283507): Create a regression test for RFE 4287690

### Reviewers
 * [Sergey Bylokhov](https://openjdk.java.net/census#serb) (@mrserb - **Reviewer**) ⚠️ Review applies to cba696d7d892714e974e894006c41353c425cf68
 * [Alexey Ivanov](https://openjdk.java.net/census#aivanov) (@aivanov-jdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7906/head:pull/7906` \
`$ git checkout pull/7906`

Update a local copy of the PR: \
`$ git checkout pull/7906` \
`$ git pull https://git.openjdk.java.net/jdk pull/7906/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7906`

View PR using the GUI difftool: \
`$ git pr show -t 7906`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7906.diff">https://git.openjdk.java.net/jdk/pull/7906.diff</a>

</details>

### CODE REVIEW

**Comment by manukumarvs** [comment]
Any volunteers please?.

**Comment by manukumarvs** [comment]
Any volunteers ?

**Comment by manukumarvs** [comment]
@aivanov-jdk Can you please review this.

**Review by mrserb — APPROVED** [review]

File: test/jdk/javax/swing/JComboBox/JComboBoxPopupMenuEventTest.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+                popupMenuVisibleLatch.countDown();
+                comboBox.removeAllItems();
+                String text = searchTextField.getText().trim();
+                List<String> comps = Arrays.stream(compStrs)
+                                           .filter(compStr -> compStr
+                                                   .startsWith(text))
+                                           .collect(Collectors.toList());
+                for (String comp : comps) {
+                    comboBox.addItem(comp);
+                }
```
Comment (aivanov-jdk) [inline test/jdk/javax/swing/JComboBox/JComboBoxPopupMenuEventTest.java:166]: Just a suggestion: use `.forEach` and add items to the `comboBox` right in the stream without creating an intermediate list.

**Review by aivanov-jdk — APPROVED** [review]
Looks good.

**Review by aivanov-jdk — APPROVED** [review]

**Pull request closed without merging**